api/envoy/api/v2/cds.proto

```diff
@@ -632,6 +641,33 @@ message Cluster {
   // If this flag is not set to true, Envoy will wait until the hosts fail active health
   // checking before removing it from the cluster.
   bool drain_connections_on_host_removal = 32;
+
+  // New mechanism for LB policy configuration. Used only if the
+  // :ref:`lb_policy<envoy_api_field_Cluster.lb_policy>` field has the value
+  // :ref:`NEW_POLICY_CONFIG<envoy_api_enum_value_Cluster.LbPolicy.NEW_POLICY_CONFIG>`.
+  LoadBalancingPolicy load_balancing_policy = 40;
+}
+
+// New Load balancing policy selection.
+//
+// Every LB policy defined via this mechanism will be identified via a
+// unique name using reverse DNS notation. If the policy needs configuration
+// parameters, it must define a message for its own configuration, which will
+// be stored in the config field. The name of the policy will tell clients
+// which type of message they should expect to see in the config field.
+message LoadBalancingPolicy {
+  message Policy {
+    // Required. The name of the LB policy.
+    string name = 1;
+    // Optional config for the LB policy.
+    // No more than one of these two fields may be populated.
+    google.protobuf.Struct config = 2;
+    google.protobuf.Any typed_config = 3;
```

Comment:
question: why even bother with struct? why not always have ANY? esp when the filter level struct is being deprecated IIRC

Reply:
ignore

Reply:
For v2, we want to stick with Struct everywhere. v3 will have Any-only (and Struct-in-Any for those who need it).